tests/core/test_test.py

```diff
@@ -1340,18 +1340,22 @@ def test_generate_input_data_using_sql(mocker: MockerFixture, tmp_path: Path) ->
   inputs:
     sqlmesh_example.incremental_model:
       query: |
-        SELECT 1 AS id, 1 AS item_id
+        SELECT 1 AS item_id, 1 AS id
         UNION ALL
-        SELECT 2 AS id, 1 AS item_id
+        SELECT 1 AS item_id, 2 AS id
         UNION ALL
         SELECT 3 AS id, 2 AS item_id
+        UNION ALL
+        SELECT 3 as item_id
```

Comment:
Can we also add:
```
UNION ALL
SELECT 4 as item_id
```

Reply:
At the same time, I don't get it. This looks like a badly formed SQL. Why support this at all?

Reply:
Yes, that's true this is a poorly formed SQL in this test, but it was meant to ensure all select arrangements work. For example, currently if we do `SELECT 1 as item_id` this would generate `SELECT 1 as id, null as item_id,` so this fix aims to address this and produce the correct sql when the order is random.

So the main purpose is to support skipping columns, when a table has many and the user only needs to test a few, so that they don't have to explicitly define all columns in the right order.

Reply:
Yeah not sure if we want to support this pattern, it may make the implementation more complex. Let's just deal with reasonable inputs. I think one fair expectation is that if you skip some columns in one query of a union, you'll skip it in all of them.

Reply:
Ok adapted the logic to have this expectation from the user